lib/traffic_controller/client.rb

```diff
@@ -76,31 +77,36 @@ def initialize(body:, boundary:)
       end
 
       def next_part
-        # trafficcontroller does not put headers in the multipart body, so we are not
-        # going to parse part headers
-        delim, _header, body = chunks.next
-        if delim == boundary_delimiter
-          return body
-        end
-        nil
-      rescue StopIteration
+        @chunks ||= parse(@body, @boundary)
+        @chunks.next
+      rescue StopIteration, ParseError
```

Comment:
@bradylove why is the rescue on `ParseError` needed? Looks like the code treats `truncated multipart message` as a cosmetic error, was this intended?

Reply:
We tried in this PR to be as minimally invasive as possible. The `ParseError` attempts to document to a future reader the multipart body was not formatted correctly. If the caller of the `MultipartParser` wanted to handle this error differently, they would at least have a clear place to start.

Reply:
We'll discuss with team consuming loggregator's gRPC API.